test/framework/filetools.py

```diff
@@ -494,6 +494,53 @@ def test_guess_patch_level(self):
         ]:
             self.assertEqual(ft.guess_patch_level([patched_file], self.test_buildpath), correct_patch_level)
 
+    def test_back_up_file(self):
+        """Test back_up_file function."""
+        fh, fp = tempfile.mkstemp()
+        os.close(fh)
+
+        ft.write_file(fp, 'foobar')
+
+        # Test simple file backup
+        ft.back_up_file(fp)
```

Comment:
maybe check with `os.listdir` that you now only have two files, the original and the backup, nothing else?